slatedb/src/tablestore.rs

```diff
@@ -87,25 +91,63 @@ impl TableStore {
     }
 
     pub fn new_with_fp_registry(
-        object_store: Arc<dyn ObjectStore>,
+        main_object_store: Arc<dyn ObjectStore>,
+        wal_object_store: Option<Arc<dyn ObjectStore>>,
         sst_format: SsTableFormat,
         path_resolver: PathResolver,
         fp_registry: Arc<FailPointRegistry>,
         block_cache: Option<Arc<dyn DbCache>>,
     ) -> Self {
         Self {
-            object_store: object_store.clone(),
+            main_object_store: main_object_store.clone(),
+            wal_object_store: wal_object_store.clone(),
```

Comment:
Random thought: An alternative approach would be to `.or_else(main_object_store.clone())` and always have a `wal_object_store`. Might get rid of some of the `for_wal` stuff below.